src/spark_rapids_ml/core.py

```diff
@@ -732,7 +748,28 @@ def predict_udf(iterator: Iterator[pd.DataFrame]) -> Iterator[pd.Series]:
 
                 yield cuml_transform_func(cuml_object, data)
 
-        return dataset.withColumn(pred_name, predict_udf(struct(*select_cols)))
+        pred_name = self.getOrDefault(self.predictionCol)
+        pred_col = predict_udf(struct(*select_cols))
+
+        if self._is_single_pred(dataset.schema):
+            return dataset.withColumn(pred_name, pred_col)
+        else:
+            # Avoid same naming. `echo sparkcuml | base64` = c3BhcmtjdW1sCg==
+            pred_struct_col_name = "_prediction_struct_c3BhcmtjdW1sCg=="
```

Comment:
Just curious if we really need a base64 string suffix here (since it's essentially a constant)?  Could it just be a well-known suffix, e.g. "_prediction_struct_multi"?

Reply:
Hmm, we can.  But it, I think, can reduce the same naming probability if we add a base64 string suffix